.github/workflows/build.yml

```diff
@@ -0,0 +1,110 @@
+name: Build and Test
+
+on:
+  push:
+    branches: [ master, main, develop, copilot/** ]
+  pull_request:
+    branches: [ master, main, develop ]
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  build-linux:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        compiler: [gcc, clang]
+    
+    steps:
+    - uses: actions/checkout@v4
+    
+    - name: Install dependencies
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y cmake build-essential
+    
+    - name: Configure CMake (GCC)
+      if: matrix.compiler == 'gcc'
+      run: |
+        mkdir build
+        cd build
+        cmake .. -DCMAKE_BUILD_TYPE=Release
+    
+    - name: Configure CMake (Clang)
+      if: matrix.compiler == 'clang'
+      run: |
+        mkdir build
+        cd build
+        CC=clang CXX=clang++ cmake .. -DCMAKE_BUILD_TYPE=Release
+    
+    - name: Build
+      run: |
+        cd build
+        cmake --build . --config Release
```

Comment:
See if you could use the modern calling of build in a single line

Reply:
@echoix Used the modern single line call in this commit: https://github.com/DRSDavidSoft/pxlib/pull/1/changes/7bc26df965afada5c7cf5ff2457a6c8c9a921edd